src/cmd/run.js

```diff
@@ -99,16 +97,19 @@ export function defaultAddonReload(
 
 
 export type ExitParams = {
+   firefox: FirefoxProcess,
+   stdin: tty$ReadStream,
    logger?: typeof log,
-   runningFirefox: FirefoxProcess,
  };
 
 export async function defaultExitProgram(
-  {runningFirefox, logger = log}: ExitParams = {},
+  {
+    firefox, stdin, logger = log,
+  }: ExitParams = {},
 ): Promise<void> {
-  logger.info('Exiting web-ext on user request');
-  userExitRequest = true;
-  runningFirefox.kill();
+  log.info('\nExiting web-ext on user request');
+  firefox.kill();
+  stdin.pause();
```

Comment:
Since it is not a `process.exit()` anymore I decided to place it here instead. Without `firefox.kill();` pause wouldn't work